pkg/lib/dynatrace_test.go

```diff
@@ -0,0 +1,136 @@
+package lib
+
+import (
+	"bytes"
+	"net/http"
+	"os"
+	"testing"
+
+	"github.com/keptn-contrib/dynatrace-service/pkg/credentials"
+	keptncommon "github.com/keptn/go-utils/pkg/lib/keptn"
+	keptnv2 "github.com/keptn/go-utils/pkg/lib/v0_2_0"
+)
+
+func TestDynatraceHelper_createClient(t *testing.T) {
+
+	mockTenant := "https://mySampleEnv.live.dynatrace.com"
+	mockReq, err := http.NewRequest("GET", mockTenant+"/api/v1/config/clusterversion", bytes.NewReader(make([]byte, 100)))
+	if err != nil {
+		t.Errorf("DynatraceHelper.createClient(): unable to make mock request: error = %v", err)
+		return
+	}
+
+	mockProxy := "https://proxy:8080"
+	t.Logf("Using mock proxy: %v", mockProxy)
+
+	type proxyEnvVars struct {
+		httpProxy  string
+		httpsProxy string
+		noProxy    string
+	}
+	type fields struct {
+		DynatraceCreds     *credentials.DTCredentials
+		Logger             keptncommon.LoggerInterface
+		OperatorTag        string
+		KeptnHandler       *keptnv2.Keptn
+		KeptnBridge        string
+		configuredEntities *ConfiguredEntities
+	}
+	type args struct {
+		req *http.Request
+	}
+
+	// only one test can be run in a single test run due to the ProxyConfig environment being cached
+	// see envProxyFunc() in transport.go for details
+	tests := []struct {
+		name         string
+		proxyEnvVars proxyEnvVars
+		fields       fields
+		args         args
+		wantErr      bool
+		wantProxy    string
+	}{
+		{
+			name: "testWithProxy",
+			proxyEnvVars: proxyEnvVars{
+				httpProxy:  mockProxy,
+				httpsProxy: mockProxy,
+				noProxy:    "localhost",
+			},
+			fields: fields{
+				DynatraceCreds: &credentials.DTCredentials{
+					Tenant:   mockTenant,
+					ApiToken: "",
+				},
+				Logger: keptncommon.NewLogger("", "", ""),
+			},
+			args: args{
+				req: mockReq,
+			},
+			wantProxy: mockProxy,
+		},
+		/*{
+			name: "testWithNoProxy",
+			fields: fields{
+				DynatraceCreds: &credentials.DTCredentials{
+					Tenant:   mockTenant,
+					ApiToken: "",
+				},
+				Logger: keptncommon.NewLogger("", "", ""),
+			},
+			args: args{
+				req: mockReq,
+			},
+			wantProxy: "",
+		},*/
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+
+			os.Setenv("HTTP_PROXY", tt.proxyEnvVars.httpProxy)
```

Comment:
In the Readme.md, please introduce a (sub)-section that explains that a user can use a proxy using the environment variables `HTTP_PROXY` etc. You can also reference the Go docs, e.g. https://golang.org/pkg/vendor/golang.org/x/net/http/httpproxy/

Furhtermore, we should allow for setting these proxy values in the values.yaml of the Helm chart.